src/components/input-number/nz-input-number.component.ts

```diff
@@ -127,22 +127,25 @@ export class NzInputNumberComponent implements ControlValueAccessor {
     return this._step;
   }
 
-  @Output() nzBlur: EventEmitter<MouseEvent> = new EventEmitter();
-  @Output() nzFocus: EventEmitter<MouseEvent> = new EventEmitter();
+  // TODO: should reconsider the payload
+  @Output() nzBlur: EventEmitter<FocusEvent | KeyboardEvent> = new EventEmitter();
+  @Output() nzFocus: EventEmitter<FocusEvent> = new EventEmitter();
```

Comment:
The original type was wrong, it will emit either `FocusEvent` or `KeyboardEvent` in runtime. Should reconsider what is the desired payload.

Reply:
should
```typescript
  @Output() nzBlur: EventEmitter<FocusEvent> = new EventEmitter();
  @Output() nzFocus: EventEmitter<FocusEvent> = new EventEmitter();
```

```typescript
_emitKeydown($event) {
    if ($event.keyCode === TAB && this._focused) {
      this._mouseInside = false;
      this._inputNumber.nativeElement.blur();
    }
  }
```

Reply:
My suggestion is not emit anything, the events themselves are enough for consuming the `blur`/`focus` change. (Or unify them into one observable with type `boolean`).